src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabHookCreator.java

```diff
@@ -104,8 +122,35 @@ public static void register(GitLabSCMSource source,
             LOGGER.log(Level.WARNING,
                     "Could not manage project hooks for " + source.getProjectPath() + " on " + server.getServerUrl(), e);
         }
+        switch (systemhookMode) {
```

Comment:
This will never be reached if I have webhook disabled but enabled systemhook :)

Reply:
Fixed in 8bc8f46